src/v/datalake/tests/record_multiplexer_bench.cc

```diff
@@ -0,0 +1,564 @@
+  record_multiplexer_bench_fixture, avro_385_byte_message_linear_31_fields) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
+    static constexpr ::testing::avro_generator_config gen_config = {
+      .string_length_range{9, 9}};
+
+    static std::string avro_schema = generate_linear_avro(31);
+
+    static thread_local chunked_vector<model::record_batch> batch_data
+      = co_await generate_avro_batches(
+        records_per_batch, batches, "avro_schema", avro_schema, gen_config);
+
+    auto reader = model::make_fragmented_memory_record_batch_reader(
+      share_batches(batch_data));
+
+    auto consumer = counting_consumer{.mux = create_mux()};
+
+    perf_tests::start_measuring_time();
+    auto res = co_await reader.consume(std::move(consumer), model::no_timeout);
+    perf_tests::stop_measuring_time();
+
+    co_return res.total_bytes;
+}
+
+PERF_TEST_CN(
+  record_multiplexer_bench_fixture, avro_385_byte_message_linear_62_fields) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
+    static constexpr ::testing::avro_generator_config gen_config = {
+      .string_length_range{4, 4}};
+
+    static std::string avro_schema = generate_linear_avro(62);
+
+    static thread_local chunked_vector<model::record_batch> batch_data
+      = co_await generate_avro_batches(
+        records_per_batch, batches, "avro_schema", avro_schema, gen_config);
+
+    auto reader = model::make_fragmented_memory_record_batch_reader(
+      share_batches(batch_data));
+
+    auto consumer = counting_consumer{.mux = create_mux()};
+
+    perf_tests::start_measuring_time();
+    auto res = co_await reader.consume(std::move(consumer), model::no_timeout);
+    perf_tests::stop_measuring_time();
+
+    co_return res.total_bytes;
+}
+
+PERF_TEST_CN(
+  record_multiplexer_bench_fixture, avro_385_byte_message_nested_31_levels) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
+    static constexpr ::testing::avro_generator_config gen_config = {
+      .string_length_range{9, 9}, .max_nesting_level = 40};
+
+    static std::string avro_schema = generate_nested_avro(31);
+
+    static thread_local chunked_vector<model::record_batch> batch_data
+      = co_await generate_avro_batches(
+        records_per_batch, batches, "avro_schema", avro_schema, gen_config);
+
+    auto reader = model::make_fragmented_memory_record_batch_reader(
+      share_batches(batch_data));
+
+    auto consumer = counting_consumer{.mux = create_mux()};
+
+    perf_tests::start_measuring_time();
+    auto res = co_await reader.consume(std::move(consumer), model::no_timeout);
+    perf_tests::stop_measuring_time();
+
+    co_return res.total_bytes;
+}
+
+PERF_TEST_CN(
+  record_multiplexer_bench_fixture, avro_385_byte_message_nested_62_levels) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
```

Comment:
nit: is the intent that when running this test suite as a whole that you compare the results of the individual cases? If so, would it make sense to have these be in an anonymous namespace and used by each case? They seem to be all the as is